agent/utils/utils_test.go

```diff
@@ -203,3 +203,27 @@ func TestParseBool(t *testing.T) {
 		}
 	}
 }
+
+func TestParseInt(t *testing.T) {
+	default_ := 5
+
+	checker := func(str string, default_ int, shouldErr bool, expectedVal int) {
+		observedVal, observedErr := ParseInt(str, default_)
+		if observedErr != nil {
+			if !shouldErr {
+				t.Fatalf("Expected err when parsing '%s' to int with default %d", str, default_)
+			}
+		} else {
+			if observedVal != expectedVal {
+				t.Fatalf("Expected %d when parsing '%s' to int with default %d", expectedVal, str, default_)
+			}
+		}
+	}
+
+	checker("1", default_, false, 1)
```

Comment:
minor: I guess it's fine in this case because`checker` is doing basic validation. But, generally, I'm in favor of a Test doing/testing 1 thing. Can you split this into multiple tests?